### PR TITLE
Tsc 214 column info remove disable checkbox and change color specifically for blocks

### PR DESCRIPTION
**Pull request by JacquiFreeElectron**

### CODE REVIEW

File: app/src/SettingsTabs/Column.tsx [hunk 1]
```diff
@@ -63,30 +64,62 @@ const ColumnAccordion: React.FC<ColumnAccordionProps> = observer(
...
+          </>
+        );
+      }
+
     }
 
     // if there are no children, don't make an accordion
     if (!hasChildren) {
-      return checkbox("column-leaf");
+      return checkbox("column-leaf", disable);
```
Comment (paologum) [inline app/src/SettingsTabs/Column.tsx:122]: remove disable

File: app/src/util/util.ts [hunk 2]
```diff
@@ -7,5 +7,18 @@
...
+    if (userBaseAge == userTopAge) {
+        return true
+    }
+    if ((minDataAge == 99999 && maxDataAge == -99999) || (minDataAge == 0 && maxDataAge == 0)) {
+        return false
+    }
+
+    if (minDataAge == userTopAge || maxDataAge == userBaseAge) {
+        return false
+    }
```
Comment (paologum) [inline app/src/util/util.ts:19]: hmm why do you have this check here? i don't believe this is true. if `minDatAge` equals the `userTopAge` but `maxDataAge < userBaseAge` it is enabled. Isn't this already caught by the former if statement on line 23

File: app/src/util/util.ts [hunk 3]
```diff
@@ -7,5 +7,18 @@
...
+    if (minDataAge <= userTopAge && maxDataAge >= userBaseAge) {
+        return false
```
Comment (paologum) [inline app/src/util/util.ts:21]: isn't this already caught by the `if` statement on line 23 as well?

File: shared/src/index.ts [hunk 4]
```diff
@@ -187,35 +187,35 @@ export type FontsInfo = {
...
+  "Ruler Tick Mark Label": { inheritable: false },
+  "Sequence Column Label": { inheritable: false },
+  "Uncertainty Label": { bold: false, color: "#000000", fontFace: "Arial", inheritable: false, italic: false, size: 5 },
+  "Zone Column Label": { bold: false, color: "#000000", fontFace: "Arial", inheritable: false, italic: false, size: 12 }
 }
 
 export type SubBlockInfo = {
   label: string,
   age: number,
   info: string,
```
Comment (paologum) [inline shared/src/index.ts:217]: this changed to `popup` no?

File: shared/src/index.ts [hunk 5]
```diff
@@ -418,8 +418,8 @@ export function assertSubBlockInfo(o: any): asserts o is SubBlockInfo {
     throwError("SubBlockInfo", "age", "number", o.number)
   if (typeof o.info !== "string")
```
Comment (paologum) [inline shared/src/index.ts:419]: `popup`

**Review by paologum — CHANGES_REQUESTED** [review]
some comments

**Comment by JacquiFreeElectron** [comment]
https://github.com/earthhistoryviz/tsconline/assets/104879321/3f2aeb7f-8757-451d-861c-5c8c35fafef2

File: app/src/SettingsTabs/Column.tsx [hunk 6]
```diff
@@ -66,60 +66,42 @@ const ColumnAccordion: React.FC<ColumnAccordionProps> = observer(
...
-                />
-                {columnName}
-              </div>
-            </ColumnContainer>
-          </>
-        );
+      let disableCheckBox = !dataInrange;
+      if (details.name === "Ma" || details.name === "Root") {
+        disableCheckBox = false;
       }
```
Comment (paologum) [inline app/src/SettingsTabs/Column.tsx:76]: reduce this to just `let disableCheckbBOx = !dataInRange || details.name === "MA" || details.name === "Root"`

File: app/src/SettingsTabs/Column.tsx [hunk 7]
```diff
@@ -66,60 +66,42 @@ const ColumnAccordion: React.FC<ColumnAccordionProps> = observer(
...
+            <div
+              className={"column-checkbox " + leaf}
+              onClick={() => clickColumnName()}
+            >
+              <TSCCheckbox
+                checked={checkedOn}
+                disabled={disableCheckBox}
+                onChange={() => {
+                  actions.toggleSettingsTabColumn(details.name);
+                }}
```
Comment (paologum) [inline app/src/SettingsTabs/Column.tsx:90]: you should conditionally load this to have either a checkbox or an error icon with a tooltip attached

**Review by paologum — CHANGES_REQUESTED** [review]
tooltip and error icon

**Comment by JacquiFreeElectron** [comment]
Screenshot 2024-02-25 at 12 18 12 AM

File: app/src/SettingsTabs/Column.tsx [hunk 8]
```diff
@@ -73,15 +107,19 @@ const ColumnAccordion: React.FC<ColumnAccordionProps> = observer(
             >
               <TSCCheckbox
                 checked={details.on}
+                //disabled={disableCheckBox}
```
Comment (paologum) [inline app/src/SettingsTabs/Column.tsx:110]: remove this

File: app/src/util/util.ts [hunk 9]
```diff
@@ -7,5 +7,16 @@
  * @returns 
  */
 export function checkIfDataIsInRange(minDataAge: number, maxDataAge: number, userTopAge: number, userBaseAge: number) {
-    return !((minDataAge > userTopAge && minDataAge < userBaseAge) || (maxDataAge < userBaseAge && maxDataAge > userTopAge))
+    if (userBaseAge == userTopAge) {
+        return false
+    }
+    if ((minDataAge == 99999 && maxDataAge == -99999) || (minDataAge == 0 && maxDataAge == 0)) {
+        return true
+    }
```
Comment (paologum) [inline app/src/util/util.ts:15]: just a comment here but once we finish datapack parsing we should not have any at 99999 and -99999

File: app/src/SettingsTabs/Column.tsx [hunk 10]
```diff
@@ -63,7 +66,38 @@ const ColumnAccordion: React.FC<ColumnAccordionProps> = observer(
...
     );
+
+
+    const dataInrange = checkIfDataIsInRange(details.minAge, details.maxAge, state.settings.topStageAge, state.settings.baseStageAge)
+
     function checkbox(leaf: string) {
+
+      let generateErrorIcon = !dataInrange && !(details.name === "Ma" || details.name === "Root")
+
+      if (generateErrorIcon) {
```
Comment (paologum) [inline app/src/SettingsTabs/Column.tsx:77]: you can shorten this to just have `!dataInrange && !(details.name === "Ma" || details.name === "Root")` in the if statement unless you use it somewhere else

**Review by paologum — CHANGES_REQUESTED** [review]

**Comment by JacquiFreeElectron** [comment]
Screenshot 2024-02-25 at 6 25 19 PM

File: app/src/SettingsTabs/Column.tsx [hunk 11]
```diff
@@ -60,7 +64,48 @@ const ColumnAccordion: React.FC<ColumnAccordionProps> = observer(
...
+                    style={{
+                      color: theme.palette.error.main,
+                      paddingLeft: 9, paddingTop: 7, paddingRight: 9
+                    }}
+                  />
+                </Tooltip>
+
+                {columnName}
+              </div>
+            </ColumnContainer >
```
Comment (paologum) [inline app/src/SettingsTabs/Column.tsx:105]: you're condition are only changing `Tooltip` or `TSCCheckbox` so it should conditionally load those not the whole `ColumnContainer`

File: app/src/SettingsTabs/Column.tsx [hunk 12]
```diff
@@ -60,7 +64,48 @@ const ColumnAccordion: React.FC<ColumnAccordionProps> = observer(
...
+                className={"column-checkbox " + leaf}
+                onClick={() => clickColumnName()}
+              >
+                <Tooltip title="Data not included in time range" placement="top" slotProps={{
+                  popper: {
+                    modifiers: [
+                      {
+                        name: 'offset',
+                        options: {
+                          offset: [0, -20],
```
Comment (paologum) [inline app/src/SettingsTabs/Column.tsx:88]: can you explain why we need the offset? also can you add arrow to Tooltip

File: app/src/SettingsTabs/Column.tsx [hunk 13]
```diff
@@ -60,7 +64,48 @@ const ColumnAccordion: React.FC<ColumnAccordionProps> = observer(
...
+                        },
+                      },
+                    ],
+                  },
+                }}>
+                  <ErrorOutlineIcon
+                    fontSize={"small"}
+                    style={{
+                      color: theme.palette.error.main,
+                      paddingLeft: 9, paddingTop: 7, paddingRight: 9
```
Comment (paologum) [inline app/src/SettingsTabs/Column.tsx:98]: move to css

Comment (paologum) [reply app/src/SettingsTabs/Column.tsx:98]: also move fontSize to css too

**Review by paologum — CHANGES_REQUESTED** [review]

File: app/src/SettingsTabs/Column.tsx (hunk 12 above)
Comment (paologum) [reply app/src/SettingsTabs/Column.tsx:88]: specifically why it's needed in popper -> modifiers -> offset

**Review by paologum — APPROVED** [review]
looks great!

**Comment by JacquiFreeElectron** [comment]
Final look: Screenshot 2024-02-26 at 7 11 33 PM
